opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -297,6 +299,64 @@ type OpenSearchCluster struct {
 	Status ClusterStatus `json:"status,omitempty"`
 }
 
+// Describes the specs of an index
+type OpensearchIndexSpec struct {
+	// Configuration options for the index
+	Settings *apiextensionsv1.JSON `json:"settings,omitempty"`
+
+	// Mapping for fields in the index
+	Mappings *apiextensionsv1.JSON `json:"mappings,omitempty"`
+
+	// Aliases to add
+	Aliases map[string]OpensearchIndexAliasSpec `json:"aliases,omitempty"`
+}
+
+// Translate rewrites the CRD format to the gateway format
+func (r *OpensearchIndexSpec) Translate() requests.Index {
+	aliases := make(map[string]requests.IndexAlias)
+	for key, val := range r.Aliases {
+		aliases[key] = requests.IndexAlias{
+			Index:        val.Index,
+			Alias:        val.Alias,
+			Filter:       val.Filter,
+			Routing:      val.Routing,
+			IsWriteIndex: val.IsWriteIndex,
+		}
+	}
+
+	request := requests.Index{}
+
+	if len(aliases) > 0 {
+		request.Aliases = aliases
+	}
+	if r.Settings.Size() > 0 {
+		request.Settings = r.Settings
+	}
+	if r.Mappings.Size() > 0 {
+		request.Mappings = r.Mappings
+	}
+
+	return request
+}
+
+// Describes the specs of an index alias
+type OpensearchIndexAliasSpec struct {
+	// The name of the index that the alias points to.
+	Index string `json:"index,omitempty"`
+
+	// The name of the alias.
+	Alias string `json:"alias,omitempty"`
+
+	// Query used to limit documents the alias can access.
+	Filter *apiextensionsv1.JSON `json:"filter,omitempty"`
+
+	// Value used to route indexing and search operations to a specific shard.
+	Routing string `json:"routing,omitempty"`
+
+	// If true, the index is the write index for the alias
+	IsWriteIndex bool `json:"isWriteIndex,omitempty"`
+}
+
```

Comment:
These types do not belong in this file, if they are common to both component and index template please put them in a separate file